lib/api_local.ml

```diff
@@ -164,17 +164,25 @@ module Slack_json : Api.Slack = struct
 end
 
 module Buildkite : Api.Buildkite = struct
-  let get_build_branch ~ctx:_ (n : Github_t.status_notification) =
+  let get_job_log ~ctx:_ _build = failwith "Not implemented"
+
+  let get_build ~ctx:_ (n : Github_t.status_notification) =
     match n.target_url with
     | None -> Lwt.return_error "no build url. Is this a Buildkite notification?"
     | Some build_url ->
     match Re2.find_submatches_exn Util.Build.buildkite_org_pipeline_build_re build_url with
     | exception _ -> failwith "Failed to parse Buildkite build url"
     | [| Some _; Some org; Some pipeline; Some build_nr |] ->
       let file = clean_forward_slashes (sprintf "organizations/%s/pipelines/%s/builds/%s" org pipeline build_nr) in
-      let url = Filename.concat buildkite_cache_dir file in
-      with_cache_file url (fun s : Github_t.branch ->
-          let { branch; _ } : Buildkite_t.get_build_response = Buildkite_j.get_build_response_of_string s in
-          { name = branch })
+      if not (Sys.file_exists file) then Lwt.return_error (sprintf "No file %s found." file)
+      else (
+        let url = Filename.concat buildkite_cache_dir file in
+        with_cache_file url (fun s : Buildkite_t.get_build_response ->
+            Devkit.Files.save_as "builds.json" (fun file -> output_string file s);
+            Buildkite_j.get_build_response_of_string s))
     | _ -> failwith "failed to get all build details from the notification. Is this a Buildkite notification?"
+  let get_build_branch ~(ctx : Context.t) (n : Github_t.status_notification) =
+    let ( let* ) = Lwt_result.bind in
```

Comment:
i'm not fond of using this here and there. Either we use it everywhere in the codebase or nowhere. Otherwise, we're just mixing styles up.

Reply:
I think I used it in a bunch of places. What is the alternative ?